src/auction.vy

```diff
@@ -210,6 +212,22 @@ def __init__(_token: ERC20, _nft: ERC721, _fee: uint256):
     self.extension_time_seconds = 3600  # default 1 hour
 
 
+
+@external
+def mint_and_start_auction(uri: String[176], patron: address, nft_contract: NftContract):
+    """
+    @dev Mints a new NFT and starts an auction for it.
+    @param uri The URI for the NFT to be minted.
+    @param patron The address of the patron for whom the auction is being started.
+    @param nft_contract The address of the NFT contract with minting capability.
+    """
+    self._check_owner_or_auctioneer()
+    token_id: uint256 = nft_contract.mint(uri, self)  # Mint directly to the auction contract
```

Comment:
Why this uses mint instead of safe_mint?

Reply:
adjusted, normal solidity shining through 😞